api_app/analyzers_manager/observable_analyzers/hunter_how.py

```diff
@@ -0,0 +1,59 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import base64
+
+import requests
+
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerConfigurationException, AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class Hunter_How(classes.ObservableAnalyzer):
+    base_url: str = "https://api.hunter.how/search"
+
+    def set_params(self, params):
+        self.__api_key = self._secrets["api_key_name"]
+        self.base_url = "https://api.hunter.how/search"
+        if not self.__api_key:
+            raise AnalyzerConfigurationException("API key is required")
+        if self.observable_classification == self.ObservableTypes.IP:
+            self.query = f'ip="{self.observable_name}"'
+        elif self.observable_classification == self.ObservableTypes.DOMAIN:
+            self.query = f'domain="{self.observable_name}"'
+        self.encoded_query = base64.urlsafe_b64encode(
+            self.query.encode("utf-8")
+        ).decode("ascii")
+        self.page = params.get("page")
+        self.page_size = params.get("page_size")
+        self.start_time = params.get("start_time", "")
+        self.end_time = params.get("end_time", "")
+
+    def run(self):
+        try:
+            api_url = f"?api-key={self.__api_key}"
+            query_url = f"&query={self.encoded_query}"
+            page_url = f"&page={self.page}&page_size={self.page_size}"
+            time_url = f"&start_time={self.start_time}&end_time={self.end_time}"
+            params_url = api_url + query_url + page_url + time_url
+
+            response_ip = requests.get(self.base_url + params_url)
+            response_ip.raise_for_status()
+
+        except requests.RequestException as e:
+            raise AnalyzerRunException(e)
+
+        return response_ip.json()
+
+    @classmethod
+    def _monkeypatch(cls):
+        patches = [
+            if_mock_connections(
+                patch(
+                    "requests.get",
+                    return_value=MockResponse({}, 200),
```

Comment:
Sorry, I did not understand that. Can you explain a bit?

Reply:
something like this?
`return_value=MockResponse({ "list" : [] }, 200)`

Reply:
you could retrieve a real result from the real server and set it as a mock. in that way it would a real mock because the external service does not answer empty to your request

Reply:
I added the "list" result as a mock. But I was not able to check the results. I again ran into an error (without changes made in the app) . I tried rebuilding but still the same.  I am trying to check logs for probable cause.
![ubuntu  Running  - Oracle VM VirtualBox 05-03-2023 19_26_27](https://user-images.githubusercontent.com/91889807/222965261-b3a1d011-e966-4ceb-b69a-c3fdea51c11d.png)
![ubuntu  Running  - Oracle VM VirtualBox 05-03-2023 19_25_55](https://user-images.githubusercontent.com/91889807/222965265-cb72e075-bdc3-44f3-81de-fd2432e2c8c8.png)